clients/src/main/java/org/apache/kafka/common/record/EosLogRecord.java

```diff
@@ -278,28 +204,107 @@ public boolean hasTimestampType(TimestampType timestampType) {
 
     public static EosLogRecord readFrom(DataInputStream input,
                                         long baseOffset,
-                                        long baseSequence,
+                                        int baseSequence,
                                         Long logAppendTime) throws IOException {
-        int size = input.readInt();
-        ByteBuffer recordBuffer = ByteBuffer.allocate(size);
-        input.readFully(recordBuffer.array(), recordBuffer.arrayOffset() + 4, size - 4);
-        recordBuffer.putInt(EosLogRecord.SIZE_OFFSET, size);
-        return new EosLogRecord(recordBuffer, baseOffset, baseSequence, logAppendTime);
+        int sizeOfBodyInBytes = Utils.readUnsignedVarInt(input);
+        ByteBuffer recordBuffer = ByteBuffer.allocate(sizeOfBodyInBytes);
+        input.readFully(recordBuffer.array(), recordBuffer.arrayOffset(), sizeOfBodyInBytes);
+        int totalSizeInBytes = Utils.bytesForUnsignedVarIntEncoding(sizeOfBodyInBytes) + sizeOfBodyInBytes;
+        return readFrom(recordBuffer, totalSizeInBytes, baseOffset, baseSequence, logAppendTime);
     }
 
     public static EosLogRecord readFrom(ByteBuffer buffer,
                                         long baseOffset,
-                                        long baseSequence,
+                                        int baseSequence,
```

Comment:
Sure. It's a bit more convenient to test using the `ByteBuffer`, but I can probably move this logic to a utility in the test case.